tests/components/eheimdigital/test_init.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for the init module."""
+
+from unittest.mock import MagicMock
+
+from eheimdigital.types import EheimDeviceType
+
+from homeassistant.components.eheimdigital import async_remove_config_entry_device
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from tests.common import MockConfigEntry
+
+
+async def test_remove_device(
+    hass: HomeAssistant,
+    eheimdigital_hub_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    device_registry: dr.DeviceRegistry,
+) -> None:
+    """Test removing a device."""
+    mock_config_entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+
+    await eheimdigital_hub_mock.call_args.kwargs["device_found_callback"](
+        "00:00:00:00:00:01", EheimDeviceType.VERSION_EHEIM_CLASSIC_LED_CTRL_PLUS_E
+    )
+    await hass.async_block_till_done()
+
+    mac_address: str = eheimdigital_hub_mock.return_value.main.mac_address
+
+    device_entry = device_registry.async_get_or_create(
+        config_entry_id=mock_config_entry.entry_id,
+        connections={(dr.CONNECTION_NETWORK_MAC, mac_address)},
+    )
+    assert device_entry is not None
+
+    assert not await async_remove_config_entry_device(
+        hass, mock_config_entry, device_entry
+    )
+
+    eheimdigital_hub_mock.return_value.devices = {}
+
+    assert await async_remove_config_entry_device(hass, mock_config_entry, device_entry)
```

Comment:
I think we should test this via the websocket like the frontend does. I think `enphase_envoy` has tests for this

Reply:
Thanks, I tried to find something and failed